pyomo/core/base/var.py

```diff
@@ -411,30 +434,52 @@ def domain(self, domain):
                 "Integers, Binary" % (domain,))
 
     @property
-    def lb(self):
-        """Return the lower bound for this variable."""
+    def lower(self):
+        """Return an expression for the variable lower bound.
+
+        This returns a (non-potentially variable) expression for the
+        variable lower bound.  This represents the tighter of the
+        current domain and the constant or expression provided to
+        setlb().  Note that the expression will NOT automatically
+        reflect changes to either the domain or the bound expression
+        (e.g., because of a call to setlb()).
+
+        """
         dlb, _ = self.domain.bounds()
         if self._lb is None:
             return dlb
         elif dlb is None:
-            return value(self._lb)
-        return max(value(self._lb), dlb)
-    @lb.setter
-    def lb(self, val):
-        raise AttributeError("Assignment not allowed. Use the setlb method")
+            return self._lb
+        # This is guaranteed by _process_bound():
+        # assert not is_potentially_variable(self._lb)
```

Comment:
You made a comment to me earlier about "commented out code causing mayhem", so I'll regurgitate that back to you here.